allure-generator/src/main/java/io/qameta/allure/environment/Allure1EnvironmentPlugin.java

```diff
@@ -25,19 +25,18 @@
     public Environment getData(final Configuration configuration,
                                final List<LaunchResults> launches) {
         final List<Environment> launchEnvironments = launches.stream().map(launch ->
-                launch.getExtra(ENVIRONMENT_BLOCK_NAME)
-                        .filter(Environment.class::isInstance)
-                        .map(Environment.class::cast)
-                        .orElse(new Environment())).collect(toList());
+                launch.getExtra(ENVIRONMENT_BLOCK_NAME))
+                .filter(Optional::isPresent).map(Optional::get)
+                .filter(Environment.class::isInstance)
+                .map(Environment.class::cast).collect(toList());
 
-        final List<EnvironmentItem> globalEnvValues = launchEnvironments.stream()
+        final Collection<EnvironmentItem> globalEnvValues = launchEnvironments.stream()
                 .flatMap(env -> env.getEnvironmentItems().stream())
-                .collect(collectingAndThen(groupingBy(EnvironmentItem::getName,
-                        reducing((EnvironmentItem e1, EnvironmentItem e2) -> {
-                            e1.getValues().addAll(e2.getValues());
-                            return e1;
-                        })), e -> e.values().stream().filter(Optional::isPresent).map(Optional::get).collect(toList())
-                ));
+                .collect(groupingBy(EnvironmentItem::getName, reducing(new EnvironmentItem(),
```

Comment:
you can create map<String, List<String>> first, then convert it to list of items